### PR TITLE
Keep the components order

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

PR ensures that the component groups are returned in the same order that the packages were loaded.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] ~If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide dist` and `./run ide watch`.~

### CODE REVIEW

File: engine/language-server/src/main/scala/org/enso/languageserver/libraries/ComponentGroupsValidator.scala [hunk 1]
```diff
@@ -45,7 +45,8 @@ final class ComponentGroupsValidator {
   ): Map[LibraryName, Either[ValidationError, ComponentGroups]] = {
     val init: Map[LibraryName, Either[ValidationError, ComponentGroups]] =
       componentGroups.map { case (k, v) => k -> Right(v) }
-    val groupsMap: mutable.Map[GroupReference, ComponentGroup] = mutable.Map()
+    val groupsMap: mutable.Map[GroupReference, ComponentGroup] =
+      mutable.HashMap()
```
Comment (JaroslavTulach) [inline engine/language-server/src/main/scala/org/enso/languageserver/libraries/ComponentGroupsValidator.scala:49]: Is `HashMap` appropriate here? I assume it cannot guarantee order, but I am not sure whether keeping order is needed here.

File: engine/runtime/src/main/scala/org/enso/compiler/PackageRepository.scala [hunk 2]
```diff
@@ -187,24 +188,25 @@ object PackageRepository {
...
+      * It should be modified and read only from within synchronized sections.
       * The component mapping is added to the collection after ensuring that the
       * corresponding library was loaded.
       */
     private val loadedComponents
-      : collection.concurrent.TrieMap[LibraryName, ComponentGroups] = {
+      : collection.mutable.Map[LibraryName, ComponentGroups] = {
       val builtinsName = LibraryName(Builtins.NAMESPACE, Builtins.PACKAGE_NAME)
-      collection.concurrent.TrieMap(builtinsName -> ComponentGroups.empty)
+      collection.mutable.LinkedHashMap(builtinsName -> ComponentGroups.empty)
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/scala/org/enso/compiler/PackageRepository.scala:198]: Replacing _concurrent_ with _mutable_. Is concurrent access still safe? Or is concurrent access prevented? I see bunch of `this.synchronized` below, so I assume the answer is yes, concurrent access is still safe.

**Review by JaroslavTulach — APPROVED** [review]

File: engine/runtime/src/main/scala/org/enso/compiler/PackageRepository.scala (hunk 2 above)
Comment (4e6) [reply engine/runtime/src/main/scala/org/enso/compiler/PackageRepository.scala:198]: Yes, the usages are synchronized, and there's a  comment there

File: engine/language-server/src/main/scala/org/enso/languageserver/libraries/ComponentGroupsValidator.scala (hunk 1 above)
Comment (4e6) [reply engine/language-server/src/main/scala/org/enso/languageserver/libraries/ComponentGroupsValidator.scala:49]: Hash map here doesn't break the order

Comment (JaroslavTulach) [reply engine/language-server/src/main/scala/org/enso/languageserver/libraries/ComponentGroupsValidator.scala:49]: According to [stackoverflow](https://stackoverflow.com/questions/10113448/scala-hashmap-iterate-in-insertion-order) _In general, a Scala HashMap does not guarantee the original order._.

**Pull request merged**